lib/Sema/CSWalker.cpp

```diff
@@ -10,6 +19,57 @@ class LocalCSWalker : public glu::ast::ASTWalker<LocalCSWalker, void> {
 public:
     LocalCSWalker(ScopeTable *scope) : _cs(scope) { }
     ~LocalCSWalker() { _cs.resolveConstraints(); }
+
+    /// @brief Visits a literal expression and generates type constraints.
+    void postVisitLiteralExpr(glu::ast::LiteralExpr *node)
+    {
+        // Get the literal value
+        auto value = node->getValue();
+
+        // Get the current type of the literal expression
+        glu::types::TypeBase *nodeType = node->getType();
+
+        // Create appropriate default type based on literal type
+        glu::types::TypeBase *defaultType = nullptr;
+
+        std::visit(
+            [&](auto &&arg) {
+                using T = std::decay_t<decltype(arg)>;
+
+                if constexpr (std::is_same_v<T, llvm::APInt>) {
+                    // Integer literal - default to signed 32-bit integer
+                    defaultType = new (_cs.getAllocator())
```

Comment:
the type must be created on the AST context's allocator, not the constraint system one, as it will outlive the constraint system